plugins/woocommerce/src/Caches/OrderCountCacheService.php

```diff
@@ -31,6 +33,38 @@ final public function init() {
 		add_action( 'woocommerce_order_status_changed', array( $this, 'update_on_order_status_changed' ), 10, 4 );
 		add_action( 'woocommerce_before_trash_order', array( $this, 'update_on_order_trashed' ), 10, 2 );
 		add_action( 'woocommerce_before_delete_order', array( $this, 'update_on_order_deleted' ), 10, 2 );
+		add_action( 'woocommerce_order_count', array( $this, 'refresh_cache' ) );
```

Comment:
It may be helpful to rename the hook `woocommerce_order_count` to something that make it more clear that it is refreshing the cache.

Reply:
Renamed this to `woocommerce_refresh_order_count_cache`.